gtests/test_nopython_callprofiler.cpp

```diff
@@ -212,5 +212,86 @@ TEST_F(CallProfilerTest, cancel)
+    int example_total_time_pos2 = serializeExample.find(total_time_str, example_total_time_pos1 + 1);
+    int example_total_time_pos3 = serializeExample.find(total_time_str, example_total_time_pos2 + 1);
+    int example_caller_name_pos0 = serializeExample.find(caller_name_str);
+    int example_caller_name_pos1 = serializeExample.find(caller_name_str, example_caller_name_pos0 + 1);
+    int example_caller_name_pos2 = serializeExample.find(caller_name_str, example_caller_name_pos1 + 1);
+    int example_caller_name_pos3 = serializeExample.find(caller_name_str, example_caller_name_pos2 + 1);
+    int example_call_count_pos0 = serializeExample.find(call_count_str);
+    int example_call_count_pos1 = serializeExample.find(call_count_str, example_call_count_pos0 + 1);
+    int example_call_count_pos2 = serializeExample.find(call_count_str, example_call_count_pos1 + 1);
+    int example_call_count_pos3 = serializeExample.find(call_count_str, example_call_count_pos2 + 1);
+
+    // Validate the serialization result except for the start_time and total_time
+    EXPECT_EQ(ss.str().substr(0, ss_start_time_pos0 + start_time_str.size()), serializeExample.substr(0, example_start_time_pos0 + start_time_str.size()));
+    EXPECT_EQ(ss.str().substr(ss_caller_name_pos0, ss_total_time_pos0 + total_time_str.size() - ss_caller_name_pos0), serializeExample.substr(example_caller_name_pos0, example_total_time_pos0 + total_time_str.size() - example_caller_name_pos0));
+    EXPECT_EQ(ss.str().substr(ss_call_count_pos0, ss_start_time_pos1 - ss_call_count_pos0), serializeExample.substr(example_call_count_pos0, example_start_time_pos1 - example_call_count_pos0));
+    EXPECT_EQ(ss.str().substr(ss_caller_name_pos1, ss_total_time_pos1 + total_time_str.size() - ss_caller_name_pos1), serializeExample.substr(example_caller_name_pos1, example_total_time_pos1 + total_time_str.size() - example_caller_name_pos1));
+    EXPECT_EQ(ss.str().substr(ss_call_count_pos1, ss_start_time_pos2 - ss_call_count_pos1), serializeExample.substr(example_call_count_pos1, example_start_time_pos2 - example_call_count_pos1));
+    EXPECT_EQ(ss.str().substr(ss_caller_name_pos2, ss_total_time_pos2 + total_time_str.size() - ss_caller_name_pos2), serializeExample.substr(example_caller_name_pos2, example_total_time_pos2 + total_time_str.size() - example_caller_name_pos2));
+    EXPECT_EQ(ss.str().substr(ss_call_count_pos2, ss_start_time_pos3 - ss_call_count_pos2), serializeExample.substr(example_call_count_pos2, example_start_time_pos3 - example_call_count_pos2));
+    EXPECT_EQ(ss.str().substr(ss_caller_name_pos3, ss_total_time_pos3 + total_time_str.size() - ss_caller_name_pos3), serializeExample.substr(example_caller_name_pos3, example_total_time_pos3 + total_time_str.size() - example_caller_name_pos3));
+    EXPECT_EQ(ss.str().substr(ss_call_count_pos3), serializeExample.substr(example_call_count_pos3));
+}
```

Comment:
Here is the test of the serialization function.
I check the output except for the start time and the total time of the functions.